src/counters.ts

```diff
@@ -0,0 +1,694 @@
+			trace(`chart file name: ${file}`);
+			if (this.collection[proc.pid]) {
+				// if we have collected data for this process then write Chart for it.
+				promises.push(this.writeChart(file, this.collection[proc.pid]));
+			}
+			if (this.includeMovingAverages) {
+				file = `${this.getFileNamePrefix(proc)}_sma_chart.png`;
+				trace(`sma chart file name: ${file}`);
+				if (this.smaOver4Collection[proc.pid]) {
+					// if we have simple moving average data computed for this process then  write Chart for it.
+					promises.push(this.writeChart(file, this.smaOver4Collection[proc.pid]));
+				}
+				file = `${this.getFileNamePrefix(proc)}_ema_chart.png`;
+				trace(`ema chart file name: ${file}`);				
+				if (this.emaOver4Collection[proc.pid]) {
+					// if we have exponential moving average data computed for this process then  write Chart for it.	
+					promises.push(this.writeChart(file, this.emaOver4Collection[proc.pid]));
+				}
+			}
+		});
+		trace(`waiting for all chart files to be written out ...`);
+		await Promise.all(promises);
+	}
+
+	private async writeChart(file: string, processCollection: ProcessStatisticsCollection): Promise<void> {
+		const trace = require('debug')(`${logPrefix}:writeChart:trace`);
+		trace(`processStats=${jsonDump(processCollection)}`);
+		const xKey = 'elapsed';
+		const xData: number[] = processCollection[xKey];
+		const xAxisLabel = `${xKey}(${ProcessStatisticsUnits[xKey]})`;
+		const lines: LineData[] = [];
+		const title = path.parse(file).name; //get just file name without directory paths and extension
+		Object.keys(processCollection)
+			.filter(key => key !== 'elapsed' && key !== 'timestamp' && key !== 'pid' && key !== 'ppid')
+			.forEach(key => {
+				lines.push({
+					label: `${key}(${ProcessStatisticsUnits[key]})`,
+					data: processCollection[key]
+				});
+			});
+		trace(`xData:${jsonDump(xData)}, lines:${jsonDump(lines)}, xAxisLabel: ${xAxisLabel}, file:${file}, title:${title}`);
+		await writeChartToFile(xData, lines, 'png', processCollection.timestamp[0], xAxisLabel, file, title);
+	}
+
+	private async writeCollectionData(): Promise<void> {
+		let promises: Promise<void>[] = [];
+		[Counters.TotalsProcessInfo, ...this.processesToTrack].filter(proc => this.collection[proc.pid]).forEach((proc: ProcessInfo) => {
+			let file = `${this.getFileNamePrefix(proc)}_data.json`;
+			promises.push(writeFileAsync(file, jsonDump(this.collection[proc.pid])));
+			if (this.includeMovingAverages) {
+				file = `${this.getFileNamePrefix(proc)}_sma_data.json`;
+				promises.push(writeFileAsync(file, jsonDump(this.smaOver4Collection[proc.pid])));
+				file = `${this.getFileNamePrefix(proc)}_ema_data.json`;
+				promises.push(writeFileAsync(file, jsonDump(this.emaOver4Collection[proc.pid])));
+			}
+		});
+		await Promise.all(promises);
+	}
+
+
+	private getFileNamePrefix(proc: ProcessInfo = undefined): string {
+		let file: string = path.join(this.outputDirectory, this.name);
+		if (proc) {

```

Comment:
So the default is to have the name be the entire path? That's going to be extremely long...

Reply:
Good observation, really like the diligency.
I need descriptive names so generated files can be easily identified with the test scenarios they were generated for, so I want to avoid short random filenames. If I run into long files issues I will do truncation logic, but will cross that bridge when I get there.